### PR TITLE
Microsoft defender for Cloud policy update

### PR DESCRIPTION
**Pull request by steph409**
## Overview/Summary

it is now possible to set all parameters from this policy in terraform. One of those parameters was asked for in #584 

## This PR fixes/adds/changes/removes

1. fix - there was a typo in the initiative definition (`Effect` instead of `effect`) which is why the `enableAscForKeyVault` did not work as expected.
2. there were some parameters, e.g. `enableAscForCosmosDbs`, which were not taken from `configure_management_resources`. I added all the ones which were missing.
3. updated some policy related documentation.

### Breaking Changes

NONE

## Testing Evidence

missing parameters are working now, I used the following configuration for testing:
```hcl
# Configure the management resources settings.
locals {
  configure_management_resources = {
    settings = {
      log_analytics = {
        enabled = true
        config = {
          retention_in_days                           = 30
          enable_monitoring_for_arc                   = false
          enable_monitoring_for_vm                    = false
          enable_monitoring_for_vmss                  = false
          enable_solution_for_agent_health_assessment = false
          enable_solution_for_anti_malware            = false
          enable_solution_for_azure_activity          = false
          enable_solution_for_change_tracking         = false
          enable_solution_for_service_map             = false
          enable_solution_for_sql_assessment          = false
          enable_solution_for_updates                 = false
          enable_solution_for_vm_insights             = false
          enable_solution_for_container_insights      = false
          enable_sentinel                             = false
        }
      }
      security_center = {
        enabled = true
        config = {
          email_security_contact                                = var.security_alerts_email_address
          enable_defender_for_apis                              = true
          enable_defender_for_app_services                      = true
          enable_defender_for_arm                               = true
          enable_defender_for_containers                        = false
          enable_defender_for_cosmosdbs                         = false
          enable_defender_for_cspm                              = true
          enable_defender_for_dns                               = false
          enable_defender_for_key_vault                         = true
          enable_defender_for_oss_databases                     = true
          enable_defender_for_servers                           = true
          enable_defender_for_servers_vulnerability_assessments = true
          enable_defender_for_sql_servers                       = true
          enable_defender_for_sql_server_vms                    = true
          enable_defender_for_storage                           = false
        }
      }
    }
  }
}


````

## As part of this Pull Request I have

- [X] Checked for duplicate [Pull Requests](https://github.com/Azure/terraform-azurerm-caf-enterprise-scale/pulls)
- [X] Associated it with relevant [issues](https://github.com/Azure/terraform-azurerm-caf-enterprise-scale/issues), for tracking and closure.
- [X] Ensured my code/branch is up-to-date with the latest changes in the `main` [branch](https://github.com/Azure/terraform-azurerm-caf-enterprise-scale/tree/main)
- [X] Performed testing and provided evidence.
- [X] Updated relevant and associated documentation.

### CODE REVIEW

**Comment by matt-FFFFFF** [comment]
Thanks @steph409 another good add. Will properly review and run tests tomorrow

File: modules/management/locals.tf
Comment (matt-FFFFFF) [inline modules/management/locals.tf:1]: Hi @steph409 The tests are failing, you need to update this local block to map the settings:

https://github.com/Azure/terraform-azurerm-caf-enterprise-scale/blob/967bccc691164417eac7db92432b20cec346c3f5/modules/management/locals.tf#L43-L77

**Review by matt-FFFFFF — CHANGES_REQUESTED** [review]

File: modules/management/locals.tf
Comment (steph409) [reply modules/management/locals.tf:1]: hi,
I had those changed but it seems I did not commit them, sorry for that! let me know if it is good now :)

Comment (matt-FFFFFF) [reply modules/management/locals.tf:1]: Thank you! Looks great :)

**Review by matt-FFFFFF — DISMISSED** [review]

**Review by matt-FFFFFF — APPROVED** [review]

**Comment by matt-FFFFFF** [comment]
LGTM!

**Pull request merged**